### PR TITLE
Fixes async gap handling.

### PR DESCRIPTION
**Pull request by ghost**
This fixes an issue where an async gap at the end of a stack trace would not get parsed correctly due to the trailing newline being `trim()`'d.

A couple of tests were added to cover this case.

### CODE REVIEW

File: lib/src/trace.dart [hunk 1]
```diff
@@ -146,7 +146,11 @@ class Trace implements StackTrace {
   static List<Frame> _parseVM(String trace) {
     // Ignore [vmChainGap]. This matches the behavior of
     // `Chain.parse().toTrace()`.
-    var lines = trace.trim().replaceAll(vmChainGap, '').split('\n');
+    var lines = trace
+        .trim()
+        .replaceAll(vmChainGap, '')
+        .split('\n')
+        .where((line) => (line != ''));
```
Comment (natebosch) [inline lib/src/trace.dart]: [nit] 
```suggestion
        .where((line) => line != '');
```

or

```dart
        .where((line) => line.isNotEmpty);
```

**Review by natebosch — APPROVED** [review]
Can you add a CHANGELOG entry?

File: lib/src/trace.dart (hunk 1 above)
Comment (ghost) [reply lib/src/trace.dart]: Done. Thanks

**Comment by ghost** [comment]
RE:CHANGELOG: I assume I'm supposed to bump the version and update the pubspec too, right?

**Comment by natebosch** [comment]
> I assume I'm supposed to bump the version and update the pubspec too, right?

In this case no, the existing version was OK. Anytime there is a trailing `-dev` on a version it should mean we are in between publishing, and can keep using the same version. When we get ready to publish we drop the `-dev`.

https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change

For future PRs it would also help to avoid a force push. We do a squash and merge once the PR is ready so we don't need to clean up into a single commit and force push ahead of time.

https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#handling-pull-requests

**Comment by mkustermann** [comment]
@natebosch We are a little pressed on time. Could you help us get this change published on pub and pulled into flutter/flutter? (we can do the latter)

**Comment by natebosch** [comment]
I opened https://github.com/dart-lang/stack_trace/pull/85 

Once that is approved I'll publish and update here.

**Comment by natebosch** [comment]
This is published as `1.10.0-nullsafety.3`

**Comment by ghost** [comment]
Thanks, Nate.

I see [1.10.0-nullsafety.3](https://pub.dev/packages/stack_trace/versions/1.10.0-nullsafety.3) on pub, but I noticed it has an odd `[UNIDENTIFIED]` tag on it, and [the log](https://pub.dev/documentation/stack_trace/1.10.0-nullsafety.3/log.txt) points to this being an SDK version issue.

In [the commit message](https://github.com/dart-lang/stack_trace/commit/8e6e53228758759185f4a7ae322b1451b9801624) you mentioned bumping the SDK version, but would it be possible to only bump the _patch_ version so we can stay within [currently stable SDK versions](https://dart.dev/tools/sdk/archive)?

The reason I'm asking is that `flutter update-packages --force-upgrade` doesn't seem to pick up 1.10.0-nullsafety.3, and I suspect it's to do with the SDK version.

**Comment by natebosch** [comment]
You are trying to update the version pinned in the Flutter SDK? I would not have expected the SDK version to be a problem since they should be on `2.11.0-dev` SDKs by now...

In any case I tested this on earlier SDKs and can't find an incompatibility so I'll drop the min back down.
https://github.com/dart-lang/stack_trace/pull/86

**Pull request merged**